src/Kerbalism/SteppedSim/FluxAnalysisFactory.cs

```diff
@@ -0,0 +1,223 @@
+			//var directFlux = directRawFlux;
+			//if (MainBody.hasAtmosphere && altitude < MainBody.atmosphereDepth)
+			//directFlux *= MainBody.LightTransparencyFactor(mainBodyPosition, starFlux.direction, vesselPosition, altitude);
+
+
+			// 1 standard unit Solar luminosity = 3.828e26 W
+			// Solar Irradiance (energy / unit area, ie W/m^2) relates to solar luminosity:
+			// Luminosity = 4 * pi * Irradiance * (d * d) where d = distance where irradiance was measured
+
+
+			// if we are inside the atmosphere, scale down both fluxes by the atmosphere absorbtion at the current altitude
+			// rationale : the atmosphere itself play a role in refracting the solar flux toward space, and the proportion of
+			// the emissive flux released by the atmosphere itself is really only valid when you're in space. The real effects
+			// are quite complex, this is a first approximation.
+			//
+			/*
+			if (body.hasAtmosphere && altitude < body.atmosphereDepth)
+			{
+				double atmoFactor = body.LightTransparencyFactor(altitude);
+				albedoFlux *= atmoFactor;
+				emissiveFlux *= atmoFactor;
+			}
+			*/
+			var directIrradianceSum = new NativeArray<double>(numSteps * numVessels, Allocator.TempJob);
+			var sum0 = new SumForVessel
+			{
+				numVesselsPerStep = numVessels,
+				numBodiesPerStep = numBodies,
+				input = directIrradiance,
+				output = directIrradianceSum,
+			}.Schedule(numSteps * numVessels, numVessels, bodyDirectIrradiancesJob);
+
+			var bodyCoreIrradianceSum = new NativeArray<double>(numSteps * numVessels, Allocator.TempJob);
+			var sum1 = new SumForVessel
+			{
+				numVesselsPerStep = numVessels,
+				numBodiesPerStep = numBodies,
+				input = coreIrradiance,
+				output = bodyCoreIrradianceSum,
+			}.Schedule(numSteps * numVessels, numVessels, bodyDirectIrradiancesJob);
+
+			var bodyEmissiveIrradianceSum = new NativeArray<double>(numSteps * numVessels, Allocator.TempJob);
+			var sum2 = new SumForVessel
+			{
+				numVesselsPerStep = numVessels,
+				numBodiesPerStep = numBodies,
+				input = emissiveIrradiance,
+				output = bodyEmissiveIrradianceSum,
+			}.Schedule(numSteps * numVessels, numVessels, bodyDirectIrradiancesJob);
+
+			var albedoIrradianceSum = new NativeArray<double>(numSteps * numVessels, Allocator.TempJob);
+			var sum3 = new SumForVessel
+			{
+				numVesselsPerStep = numVessels,
+				numBodiesPerStep = numBodies,
+				input = albedoIrradiance,
+				output = albedoIrradianceSum,
+			}.Schedule(numSteps * numVessels, numVessels, albedoIrradianceAtVesselJob);
+
+			var sumTemp = JobHandle.CombineDependencies(sum0, sum1, sum2);
+			var copyJob = new RecordIrradiances
+			{
+				vessels = vessels,
+				directIrradiance = directIrradianceSum,
+				albedoIrradiance = albedoIrradianceSum,
+				emissiveIrradiance = bodyEmissiveIrradianceSum,
+				coreIrradiance = bodyCoreIrradianceSum,
+			}.Schedule(numSteps * numVessels, numVessels, JobHandle.CombineDependencies(sumTemp, sum3));
+
```

Comment:
Not exactly sure where this is going, but you have no way to predict which steps will be effectively used. Therefore you can't sum their results.

Reply:
All the steps will be used.  The job isn't computing future steps that may or may not exist or be used.  It is computing up to the current Planetarium.UT.

Also, this isn't the sum across all of the timesteps, though this *might* be a good place to take a large step across.
This is the sum of all of the individual body contributions to the vessel.  And what instigated the conversation in the Discord, about if/why you might need it still per-body at the next level of processing.